### PR TITLE
Improve devcontainer setup

### PR DESCRIPTION
**Pull request by paule96**
## Description

Add the changes to the new `devcontainer.json` that were made in #3520 .

- add powershell
- add dapr
- change docker base image to ubuntu
  - This is right now the only way I know to interact with windows sshd reliabel (for example if people want to use their local ssh config)
- Use intellicode instead of devkit for C#
  -  intellicode depends on devkit so devkit is still their
- Add docs

Fixes # (issue)

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [x] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Is this introducing a breaking change?
      - [ ] Yes
        - Link to aspire-docs issue (please use this [`breaking-change` template](https://github.com/dotnet/docs-aspire/issues/new?template=04-breaking-change.yml)):
      - [ ] No
        - Link to aspire-docs issue (please use this [`doc-idea` template](https://github.com/dotnet/docs-aspire/issues/new?template=02-docs-request.yml)):
  - [x] No

cc @davidfowl & @mitchdenny 

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6570)

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -68,13 +69,13 @@
 	"otherPortsAttributes": {
 		"onAutoForward": "ignore"
 	},
-
 	// Use 'postCreateCommand' to run commands after the container is created.
 	"customizations": {
 		"vscode": {
 			"extensions": [
-				"ms-dotnettools.csdevkit",
+				"ms-dotnettools.vscodeintellicode-csharp",
```
Comment (davidfowl) [inline .devcontainer/devcontainer.json:76]: Why?

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -3,27 +3,28 @@
 {
 	"name": "C# (.NET)",
 	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
-	"image": "mcr.microsoft.com/devcontainers/dotnet:1-9.0",
+	"image": "mcr.microsoft.com/devcontainers/dotnet:1-9.0-bookworm",
```
Comment (davidfowl) [inline .devcontainer/devcontainer.json:6]: Why?

Comment (paule96) [reply .devcontainer/devcontainer.json:6]: change docker base image to ubuntu
This is right now the only way I know to interact with windows sshd reliabel (for example if people want to use their local ssh config)

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (paule96) [reply .devcontainer/devcontainer.json:76]: To sum up the discussions from the old PR:

- intellicode gives the user a better IntelliSense then without
- intelliCode depends on devkit so it's still their

File: docs/machine-requirements.md [hunk 3]
```diff
@@ -26,3 +26,22 @@ When you install, ensure that both:
 * [Windows](https://podman.io/docs/installation#windows)
 * [macOS](https://podman.io/docs/installation#macos)
 * [Linux](https://podman.io/docs/installation#installing-on-linux)
+
+## (Windows / Linux / Mac) DevContainer in VS Code
+
+On Windows you could also use VS Code with the DevContainers extension. This requieres that you have installed a container engine.
```
Comment (davidfowl) [inline docs/machine-requirements.md:32]:
```suggestion
On Windows you could also use VS Code with the DevContainers extension. This requires that you have installed a container engine.
```

File: docs/machine-requirements.md [hunk 4]
```diff
@@ -26,3 +26,22 @@ When you install, ensure that both:
...
+> :warning: This will use around 16GB of RAM, after you loaded the solution.
+
+### Install VS Code with DevContainers Extension
+
+* [VS Code](https://code.visualstudio.com/Download)
+* [DevContainers Extension](https://marketplace.visualstudio.com/items?itemName=ms-VS Code-remote.remote-containers)
+
+## (Browser) Codespaces
+
+Just start the Codespaces in your fork. The initialisation of the code space takes around 5 mins. After that you can open the solution.
```
Comment (davidfowl) [inline docs/machine-requirements.md:44]:
```suggestion
Just start the Codespaces in your fork. The initialization of the code space takes around 5 mins. After that you can open the solution.
```

File: docs/machine-requirements.md [hunk 5]
```diff
@@ -26,3 +26,22 @@ When you install, ensure that both:
...
+This will take on the free version of Codespace around 10 mins.
+
+> :warning: With the free version of Codespaces the development experience is not nice. We recommend using at least a Codespace with 16GB of RAM or use your local VS Code / DevContainers instance.
```
Comment (davidfowl) [inline docs/machine-requirements.md:47]:
```suggestion
> :warning: With the free version of Codespaces the development experience can be less than ideal. We recommend using at least a Codespace with 16GB of RAM or use your local VS Code / DevContainers instance.
```

**Comment by davidfowl** [comment]
Merging this in, I tried it out and it works well (also tried the dapr sample which might need more config to make the app ports be forwarded).

File: .devcontainer/devcontainer.json [hunk 6]
```diff
@@ -3,27 +3,28 @@
...
+		"ghcr.io/dapr/cli/dapr-cli": {
+			"version": "1.12.0"
 		}
 	},
-
 	"hostRequirements": {
 		"cpus": 8,
 		"memory": "32gb",
 		"storage": "64gb"
 	},
```
[the remark this answers is not in the document]
Comment (mitchdenny) [reply .devcontainer/devcontainer.json:25]: This is really specific to this repo. We have a few test cases that spin up a lot of containers so this is about optimizing the development experience in our repo. When .NET 9.0 ships we'll probably start experimenting with devcontainer setups outside of our repo a bit more and figure out what a good minimum config is.

[the remark this answers is not in the document]
Comment (davidfowl) [reply .devcontainer/devcontainer.json:25]: I was thinking we should ship a devcontainer item template as part of this work.

**Pull request merged**